libsignal-service/src/provisioning/manager.rs

```diff
@@ -271,22 +274,51 @@ impl<P: PushService> LinkingManager<P> {
                             })
                         })?;
 
-                    let public_key = PublicKey::deserialize(
+                    let pni_uuid = message
+                        .pni
+                        .ok_or(ProvisioningError::InvalidData {
+                            reason: "missing client UUID".into(),
+                        })
+                        .and_then(|ref s| {
+                            Uuid::parse_str(s).map_err(|e| {
+                                ProvisioningError::InvalidData {
+                                    reason: format!("invalid UUID: {}", e),
+                                }
+                            })
+                        })?;
+
+                    let aci_public_key = PublicKey::deserialize(
                         &message.aci_identity_key_public.ok_or(
                             ProvisioningError::InvalidData {
                                 reason: "missing public key".into(),
                             },
                         )?,
                     )?;
 
-                    let private_key = PrivateKey::deserialize(
+                    let aci_private_key = PrivateKey::deserialize(
                         &message.aci_identity_key_private.ok_or(
                             ProvisioningError::InvalidData {
                                 reason: "missing public key".into(),
                             },
                         )?,
                     )?;
 
+                    let pni_public_key = PublicKey::deserialize(
+                        &message.pni_identity_key_public.ok_or(
+                            ProvisioningError::InvalidData {
+                                reason: "missing public key".into(),
+                            },
+                        )?,
+                    )?;
```

Comment:
Should this indeed be a hard error? I imagine PNI is optional?

Reply:
Yeah, this is what I thought. I can try to look into it.

Reply:
No, it's always there.

Reply:
Makes sense during linking. Mergable.